fakeredis/_server.py

```diff
@@ -898,11 +908,18 @@ def _scan(self, keys, cursor, *args):
         result_cursor = cursor + count
         result_data = []
 
-        if pattern is not None:
-            regex = compile_pattern(pattern)
+        regex = compile_pattern(pattern) if pattern is not None else None
+
+        def _match_key(key):
+            return regex.match(key) if pattern is not None else True
+
+        def _match_type(key):
+            return self.type(self._db[key]).value == type if type else True
```

Comment:
Needs to be a `casematch` (or else convert `type` to lowercase when it is loaded from args). Also since I've asked above for the value of `type` not to be checked, it could be the empty string and hence `if type` should become `if type is not None`.

Reply:
Fixed.